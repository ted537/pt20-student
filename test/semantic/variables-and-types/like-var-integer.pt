using output;
var y : like 1;