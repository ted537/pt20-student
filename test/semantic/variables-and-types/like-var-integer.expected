 % output from running:
 % ssltrace "ptc -o3 -t3 -L lib/pt test/semantic/variables-and-types/like-var-integer.pt" lib/pt/semantic.def | egrep "^ *[\.o%]"
 oSymbolStkPush(syProcedure)
 oSymbolTblPushScope
 oCountPush(three)
  oSymbolStkPushLocalIdentifier
  oSymbolStkSetKind(syVariable)
  oTypeStkPush(tpFile)
  oTypeStkLinkToStandardType(stdText)
  oSymbolStkEnterTypeReference
  oValuePush(two)
   .tFileDescriptor
   oAllocateAlignOnWord
   oSymbolStkEnterDataAddress
   .tLiteralInteger
   oEmitValue
   % value emitted 2
   .tFileBind
   .tLiteralAddress
   oEmitDataAddress
   % value emitted 0
   .tStoreInteger
   oAllocateDescriptor
  oSymbolTblEnter
  oSymbolStkPop
  oTypeStkPop
  oValuePop
 oCountPop
   oSymbolStkPushLocalIdentifier
     oTypeStkPush(tpInteger)
     oTypeStkLinkToStandardType(stdInteger)
    oSymbolStkSetKind(syVariable)
     oAllocateAlignOnWord
     oSymbolStkEnterDataAddress
     oAllocateVariable
    oSymbolStkEnterTypeReference
    oSymbolTblEnter
   oTypeStkPop
   oSymbolStkPop
 oSymbolTblPopScope
 oSymbolStkPop
 .tTrapBegin
 .tTrap
 oEmitTrapKind(trHalt)
 % value emitted 0
